cub/test/catch2_test_block_run_length_decode.cu

```diff
@@ -244,6 +245,11 @@ public:
       }
 
       decoded_window_offset += DECODED_ITEMS_PER_THREAD * BLOCK_THREADS;
+      if (decoded_window_offset < decoded_size)
+      {
+        // Sync before starting the next loop, which reuses shmem
+        cub::CTA_SYNC();
+      }
```

Comment:
critical: current docs for `RunLengthDecode` state that:
> Calling __syncthreads() between any two calls to`RunLengthDecode` is not required.
If we add a sync, we'll stop verifying this guarantee. Looking at the original issue, the race is coming from the block store rather than RLD. Given that this is a test rather than a benchmark, I'd suggest to change `BlockStoreRelativeOffsetT` and `BlockStoreDecodedItemT` algorithms to use `BLOCK_STORE_DIRECT`. That'd slow down the test a bit, but we'll stop using shared memory in the loop, so sync won't be necessary.

Reply:
Nice catch! Updated the PR with the suggested change, verified that the race is no longer detected.